### ISSUE TITLE
unit test failed for float number comparision

### ISSUE THREAD

**Issue by zhaofeng-shu33** [AlanWei/deeplearning-js#2]
On my machine, the test for softmax function failed:
          Array [
    -       0.3382504271053028,
    -       0.3382504271053028,
    -       0.3382504271053028,
    +       0.33825042710530284,
    +       0.33825042710530284,
    +       0.33825042710530284,
          ],
maybe rounding to same precision and compare ?

**Comment by AlanWei** [comment]
Hi @zhaofeng-shu33 ! Thanks for submitting the issue.

Yes, I think JavaScript floating number has always been a problem for scientific calculation. When I coded deeplearning-js, I tried to use some number formatting libs to solve this issue from the root. However, all those libs slows down the whole calculation process a lot. In both forward and backward propagation, all calculation needs to be rounded to same precision. The speed is not acceptable.

It's hard to measure how this floating number problem gonna affect the training results so that I would like to put it into the future development roadmap.

BTW, are you using a Windows PC and which Node.js version you are using? All tests pass in my Mac machine right now.

<img width="468" alt="screen shot 2018-02-14 at 3 08 16 pm" src="https://user-images.githubusercontent.com/3828728/36191612-eeab33d0-1198-11e8-9711-7e3503f41010.png">
